frontend/src/pages/article/components/article-view/article-view.tsx

```diff
@@ -1,22 +1,35 @@
 import ArticleBanner from '~/assets/img/article-banner.jpg';
 import { IconButton, Tag } from '~/libs/components/components.js';
+import { useAppDispatch, useCallback, useParams } from '~/libs/hooks/hooks.js';
 import { type ArticleType } from '~/libs/types/types.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
 
 import styles from './styles.module.scss';
 
 type Properties = {
   article: ArticleType;
+  isShared: boolean;
 };
 
-const onButtonClick = (): void => {
-  /**
-   * @todo implement handle logic for buttons clicked events(favorite, comment, share)
-   */
-};
-
-const ArticleView: React.FC<Properties> = ({ article }) => {
+const ArticleView: React.FC<Properties> = ({ article, isShared }) => {
   const { title, text, tags } = article;
 
+  const { id } = useParams();
+
+  const dispatch = useAppDispatch();
+
+  const onButtonClick: React.MouseEventHandler<HTMLButtonElement> = useCallback(
+    (event): void => {
+      const button = event.currentTarget as HTMLButtonElement;
+      const iconName = button.getAttribute('name');
+
+      if (iconName === 'share' && id) {
+        void dispatch(articlesActions.shareArticle({ id }));
+      }
+    },
+    [dispatch, id],
+  );
```

Comment:
Turn back `onButtonClick` handler placeholder and use it for all actions that already used it
And use this handler specific only for `share` action
```suggestion
  const handleShareButtonClick = useCallback(
    (): void => {
      void dispatch(articlesActions.shareArticle({ id }));
    },
    [dispatch, id],
  );
```